### PR TITLE
Change default build type to empty string

### LINKED ISSUE #6961: Problems with cmake build_type variant
### Summary

The build_type variant turns on CMAKE_BUILD_TYPE, which sets optimization flags. Those flags are set after the user-specified CMAKE_C_FLAGS and equivalents. This leads to user flags being rejected under default configurations.

### Proposal

Change the default for the `build_type` variant to `""`. This is consistent with CMake defaults outside of Spack, and will allow users to specify compiler flags manually without being overridden by CMake. CMake build types can still be specified using the `build_type` variant.

### Positive effects

- CMake packages will by default respect compiler optimization flags passed by the user.
- CMake packages will now behave by default as they would outside of Spack by default.

### Potential negatives

- The default build is now less optimized
- Packages that have set a different default build type in their CMakeLists.txt file will need to override the build_type variant in their package.py file to change the default. (This is actually already true, it just now also applies to packages with default RelWithDebInfo, which was the previous Spack default).

### Technical scope

Very small. I have created a PR to change this.

### PR DESCRIPTION
**Pull request by becker33**
For compatibility with compiler flags and similarity to CMake outside of Spack

Resolves #6961. For more detail, see the issue #6961.

### CODE REVIEW

File: lib/spack/spack/build_systems/cmake.py [hunk 1]
```diff
@@ -130,9 +130,9 @@ def _std_args(pkg):
             raise InstallError(msg)
 
         try:
-            build_type = pkg.spec.variants['build_type'].value
+            build_type = pkg.spec.variants['build_type'].value or ''
```
Comment (adamjstewart) [inline lib/spack/spack/build_systems/cmake.py]: I had no idea you could do this in Python!

Comment (becker33) [reply lib/spack/spack/build_systems/cmake.py]: @scheibelp just showed it to me on a recent PR, it's great for keeping None checks clean!

**Comment by scheibelp** [comment]
Would it be worth adding a `tty.warning` somewhere in `CMakePackage` if `build_type` is set, there are compiler flags, and the flag handler uses the build system? If a package overrides `build_type` and sets their own default, they may be confused why the flags suddenly aren't set. IMO `flags_to_build_system_args` would be an appropriate place.

**Comment by becker33** [comment]
@scheibelp if we were to do that we would need to do it for all flag handlers, the bad behavior appears under all flag_handlers, it's about setting incompatible flags, in whatever way they are set.

I don't think we would want to set that warning from the build environment, because then it would be written to the build log not the terminal. I'm not sure where before that we could set it though, without seeing it every time we concretize a spec, which I think would be overkill. I don't think it's a bad idea, but I'm not sure where we could reasonably set that message.

**Comment by alalazo** [comment]
Closing as stale (similarly to the recently closed #5265)

**Pull request closed without merging**